### DISCUSSION TITLE
Grupo Independente de Pesquisa sobre E-commerce / Web Payments

### DISCUSSION THREAD

**Discussion by fititnt** [frontendbr/forum#326] [Discussão]
Relacionado ao https://github.com/webiwg/webiwg-issues/issues/28

---

O @Dkmister e eu fazemos um convite aqui para quem tem interesse formar um grupo de pesquisa sobre temas relacionados a comércio eletrônico. A participação, diferente de outros grupos, não exige vínculo com instituição de ensino, nem que e a pessoa já seja especialista na área ou trabalhe em uma empresa que paga para ela participar.

Uma das motivações é que no Brasil não há uma cultura forte como em outros países em pesquisa. Não só por falta de investimentos do governo, como dos desenvolvedores não estarem acostumados com isso, ao ponto de achar que "é coisa de quem quer fazer mestrado".

Mesmo quando há ideias geniais, o comum é isso se converter em "apenas" palestras ou postagens em sites. É raro produção de artigos científicos, e praticamente inexistente participação de brasileiros sem vínculo com governo ou universidade em grupos internacionais que definem padrões. Isso estimula brasileiros se mudarem para outros países, ou no mínimo desistirem de seguir propondo mudanças em padrões quando se sentem sozinhos ao ver que existe burocracia.

Para quem não nos conhece, pode acompanhar traduções feitas em outros dois projetos pelos links
- https://github.com/webiwg/acessibilidade-web-pt/projects/1
- https://github.com/webiwg/historia-web-pt/projects/1
## A proposta

A proposta **inicial** é reunir pessoas que **tenham interesse no tema comércio eletrônico**, mesmo que não tenham experiência, a colaborar em traduções de padrões ou artigos técnicos em outros idiomas que abordem o tema.

Normalmente quem tem experiência nessa área não vai ter interesse em participar de um grupo de pesquisa desses, se tiver interesse, não vai ter tempo. Logo, interessados em participar colaborarem na tradução de textos vai implicitamente melhorar conhecimento no tema, chamar atenção de que com ajuda podem sim dar contribuições em padrões internacionais.

Nós **não** precisamos de especialistas. Nos podemos ajudar pessoas a se **tornarem** especialistas.
## Contra partida

Não receber dinheiro diretamente não significa que não haja retorno além do conhecimento pessoal. Se houver quantidade mínima de pessoas interessadas, por menor que um grupo de pesquisa desse seja, é provável que isso se converta em pequenos entregáveis, como provas de conceito, testes com usuário e afins, que podem se converter em palestras nacionais ou internacionais, e, o que seria um dos objetivos a médio prazo, participação na definição de padrões internacionais.

Se no primeiro ano, por exemplo, o grupo já tiver feito alguns pequenos trabalhos relevantes, é uma possibilidade que como mercado de e-commerce carece de pessoas com experiência, colaboradores ativos tenham contratação na área facilitada. Empresas contratantes poderiam ser estimuladas a acordar que, assim como ocorre em outros países, ela permita por padrão que um percentual de horas do trabalho possam ser dedicadas a  participação em iniciativas como a nossa.

**Comment by suissa** [comment]
EU TENHO E MTOOO!

Estou iniciando com uns alunos uma PoC de um ecommerce baseado em eventos e com uma arquitetura híbrida q possa executar módulos escritos em qquer linguagem.

E obviamente utilizando api, com cada módulo sendo atômico e independente sempre implementando a mesma interface de REST e Eventos.

Quanto mais padrões eu souber melhor sera, pois ai com certeza eu converto em aulas para a comunidade. ;)

**Comment by fititnt** [comment]
Que tri, tchê. Acho que via esse grupo de pesquisa, pelo menos agora no início, é interessante reunir mais traduções de documentação mesmo.

Um dos motivos que faz diferença a médio prazo, até para dar "mais credibilidade" ao G.P. é tradução de especificações. Traduções não costumam tomar muito tempo, são indiscutivelmente bem aceitas e respeitadas quando bem feitas, e podem servir pra inspirar os membros.

Na lista abaixo, eu coloquei o [JSON-LD](https://www.w3.org/TR/json-ld-api/) porque parece que não temos em português, e é BEM interessante isso pra quem quer construir APIs. A versão 1.0 dele já tem versão estável, então é até mais tranquilo da gente traduzir do que sobre web payments.

Eu não coloquei nesta lista padrões relacionado a comércio eletrônico que usam XML, mas existem. Creio que como um dos objetivos das traduções é inspirar coisas como provas de conceito como você já está vendo, imaginando que ninguém aqui tem interesse de fazer APIs com XML, preferi não sugerir isso. Pelo menos não agora.

## Exemplos de recomendações que poderiam ser traduzidas

Eu não pesquisei **muito** sobre o assunto, então deve ter muito conteúdo que devo estar perdendo. Acho uma boa procurar bem antes padrões relacionados.

### Sobre web payments
- https://www.w3.org/TR/payment-method-basic-card/
- https://www.w3.org/TR/payment-method-id/
- https://www.w3.org/TR/payment-request/

### IETF
- [ECML v1.1: Field Specifications for E-Commerce](https://tools.ietf.org/html/rfc3106)
- [Electronic Commerce Modeling Language (ECML): Version 2 Requirements](https://tools.ietf.org/html/rfc3505)
- Talvez, talvez seja melhor pra G.P. sobre segurança [Javascript Object Signing and Encryption, links relacionados](https://datatracker.ietf.org/wg/jose/documents/)
- Talvez, **apenas** se o grupo tiver interesse em construir APIs, [JSON-LD 1.0 Processing Algorithms and API](https://www.w3.org/TR/json-ld-api/)

**Comment by thiamsantos** [comment]
Eu tenho interesse em contribuir e também tempo de sobra.
